zaza/openstack/utilities/openstack.py

```diff
@@ -1690,6 +1690,30 @@ def delete_image(glance, img_id):
     delete_resource(glance.images, img_id, msg="glance image")
 
 
+@tenacity.retry(wait=tenacity.wait_exponential(multiplier=1, max=60),
+                reraise=True, stop=tenacity.stop_after_attempt(5))
+def delete_volume(cinder, vol_id):
+    """Delete the given volume from cinder.
+    :param cinder: Authenticated cinderclient
+    :type cinder: cinderclient.Client
+    :param vol_id: unique name or id for the openstack resource
+    :type vol_id: str
+    """
+    delete_resource(cinder.volumes, vol_id, msg="deleting cinder volume")
+
+
+@tenacity.retry(wait=tenacity.wait_exponential(multiplier=1, max=60),
```

Comment:
resource_removed called from delete_resource is already using tenacity to retry the operation.  Why is the extra level of retrying necessary?

Reply:
removed